### ISSUE TITLE
[Enhancement]: aws_codebuild_project to support build_status_config for primary source

### ISSUE THREAD

**Issue by bob-rohan** [hashicorp/terraform-provider-aws#28324]
### Description

Please respond https://github.com/hashicorp/terraform-provider-aws/issues/27619#issuecomment-1307143463

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @bob-rohan 👋 Sorry to have missed your comment on #27619! I took a look and it looks like the resource schema [supports the `build_status_config` attribute in the `source` block](https://github.com/hashicorp/terraform-provider-aws/blob/bfcd55c7cde9b2190aa0a18e6110c967e1ad5458/internal/service/codebuild/project.go#L624-L640) as well as the `secondary_source` block, and has since version `3.38.0` of the provider (with #15442).

It looks like the documentation never got updated to include this, so I'll work on updating the documentation so that it'll hopefully be corrected by next release.
